python/build.py

```
@@ -289,6 +357,92 @@ def load_binary_tokens(path: str) -> list[int]:
+    return avg_loss, ppl
+
+
+# Fixed prompts for sampling at checkpoints (tests different capabilities)
+SAMPLE_PROMPTS = [
+    "What is the capital of France?",
+    "Explain how a neural network learns in simple terms.",
+    "Write a short poem about the ocean.",
+]
+
+
+def generate_samples(gpu_model, cfg, tokenizer, step: int,
```

Ruff (0.15.1)

[warning] Unused function argument: `step`

(ARG001)